src/Raven.Server/Documents/Sharding/Handlers/ShardedIncomingReplicationHandler.cs

```diff
@@ -26,7 +26,7 @@ public class ShardedIncomingReplicationHandler : AbstractIncomingReplicationHand
         private readonly ShardedDatabaseContext.ShardedReplicationContext _parent;
         private readonly Dictionary<int, ShardedOutgoingReplicationHandler> _handlers;
         private string _lastAcceptedChangeVector;
-        private long _lastAcceptedEtag;
+        private Dictionary<int, long> _lastSentEtagPerDestination;
```

Comment:
why we need this change?

Reply:
To avoid updating the cv in sharded destination when sending a heartbeat from the source to the wrong shard.
For example, source & destination both contain 2 shards and the same configuration -> without this change, a heartbeat message from src$0 -> dst$1 might send; then, dst$1 cv contains src$0 part, although no documents sent.

This is also a fix for https://issues.hibernatingrhinos.com/issue/RavenDB-20454/Wrong-change-vector-after-sharded-database-replication